core/src/main/java/org/jdbi/v3/core/statement/ParsedParameters.java

```diff
@@ -78,4 +78,22 @@ public String toString() {
             + ", parameterNames=" + parameterNames
             + '}';
     }
+
+    /**
+     * A static factory for {@link ParsedParameters} objects. It takes two
+     * arguments. The first argument is a boolean flag which indicates if
+     * statement, to which this object refers to, is using positional
+     * (<code>true</code> value) or named parameters (a <code>false</code>
+     * value). The second argument is a list which contains parameters
+     * available in SQL. In case when positional parameters are used the
+     * names list should contain only positional parameters, or more
+     * strictly, a "?" characters.
+     *
+     * @param positional true if statement uses positional parameters, false otherwise
+     * @param names the parameter names from SQL statement
+     * @return New {@link ParsedParameters} object
+     */
+    public static ParsedParameters of(boolean positional, List<String> names) {
```

Comment:
I think API-wise (without considering functionality yet) this factory could be better: either the params are positional and you need a List (Set? LinkedSet?) of names, or they are positional and you need to know how many (as an int) there are. I'd split this into two, more expressive methods.